### PR TITLE
apmplanner-2: 2.0.26 -> 2.0.27-rc1

### PR DESCRIPTION
**Pull request by wucke13**
###### Motivation for this change
Fixing a bug in 2.0.26, which prevents ardupilot users from changing PID values.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22709,7 +22709,7 @@ in
 
   ### SCIENCE/ROBOTICS
 
-  apmplanner2 = libsForQt59.callPackage ../applications/science/robotics/apmplanner2 { };
+  apmplanner2 = libsForQt511.callPackage ../applications/science/robotics/apmplanner2 { };
```
Comment (c0bw3b) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  apmplanner2 = libsForQt5.callPackage ../applications/science/robotics/apmplanner2 { };
```

File: pkgs/applications/science/robotics/apmplanner2/default.nix [hunk 2]
```diff
@@ -6,12 +6,12 @@
 stdenv.mkDerivation rec {
   name = "apmplanner2-${version}";
   # TODO revert Qt59 to Qt5 in pkgs/top-level/all-packages.nix on next release
```
Comment (c0bw3b) [inline pkgs/applications/science/robotics/apmplanner2/default.nix]: That comment can be removed then

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/top-level/all-packages.nix]: Nope. That's on purpose. Changing it according to your recommendation breaks the build for me:
```
build flags: -j16 -l16 SHELL=/nix/store/5p8qcz6i13ymgf80kimhx2g9pb5r5nia-bash-4.4-p23/bin/bash
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic libs/opmapcontrol/src/mapwidget/mapripform.ui -o build-release/ui/ui_mapripform.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/MainWindow.ui -o build-release/ui/ui_MainWindow.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/CommSettings.ui -o build-release/ui/ui_CommSettings.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/SerialSettings.ui -o build-release/ui/ui_SerialSettings.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/UASControl.ui -o build-release/ui/ui_UASControl.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/UASList.ui -o build-release/ui/ui_UASList.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/UASInfo.ui -o build-release/ui/ui_UASInfo.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/UASView.ui -o build-release/ui/ui_UASView.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/ParameterInterface.ui -o build-release/ui/ui_ParameterInterface.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/WaypointList.ui -o build-release/ui/ui_WaypointList.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/ObjectDetectionView.ui -o build-release/ui/ui_ObjectDetectionView.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/JoystickWidget.ui -o build-release/ui/ui_JoystickWidget.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/HDDisplay.ui -o build-release/ui/ui_HDDisplay.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/MAVLinkSettingsWidget.ui -o build-release/ui/ui_MAVLinkSettingsWidget.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/AudioOutputWidget.ui -o build-release/ui/ui_AudioOutputWidget.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/QGCSensorSettingsWidget.ui -o build-release/ui/ui_QGCSensorSettingsWidget.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/watchdog/WatchdogControl.ui -o build-release/ui/ui_WatchdogControl.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/watchdog/WatchdogProcessView.ui -o build-release/ui/ui_WatchdogProcessView.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/watchdog/WatchdogView.ui -o build-release/ui/ui_WatchdogView.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/QGCFirmwareUpdate.ui -o build-release/ui/ui_QGCFirmwareUpdate.h
/nix/store/ais74bm3fvrcj6gcx3mq2l1n51xw5gag-qtbase-5.12.0-dev/bin/uic src/ui/QGCPxImuFirmwareUpdate.ui -o build-release/ui/ui_QGCPxImuFirmwareUpdate.h
make: *** No rule to make target 'QOSGWidget.h', needed by 'build-release/ui/ui_QMap3D.h'.  Stop.
make: *** Waiting for unfinished jobs....
builder for '/nix/store/0gzs757iykcxksiahi7kccig55bp59zi-apmplanner2-2.0.27-rc1.drv' failed with exit code 2
error: build of '/nix/store/0gzs757iykcxksiahi7kccig55bp59zi-apmplanner2-2.0.27-rc1.drv' failed
```

Comment (c0bw3b) [reply pkgs/top-level/all-packages.nix]: Ok

File: pkgs/applications/science/robotics/apmplanner2/default.nix (hunk 2 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/apmplanner2/default.nix]: Done

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build apmplanner2

**Pull request merged**